cmd/olclient/gov.go

```diff
@@ -191,38 +195,49 @@ func listProposals(cmd *cobra.Command, args []string) error {
 	Ctx := NewContext()
 	fullnode := Ctx.clCtx.FullNodeClient()
 
-	if listArgs.ProposalID == "" && listArgs.State == "" {
-		return errors.New("input is empty")
+	// List single proposal
+	if listArgs.ProposalID != "" {
+		req := client.ListProposalRequest{
+			ProposalId: governance.ProposalID(listArgs.ProposalID),
+		}
+
+		reply, err := fullnode.ListProposal(req)
+		if err != nil {
+			return errors.New("error in getting proposal")
+		}
+
+		ps := reply.ProposalStats[0]
```

Comment:
You might want to create another struct for the single proposal reply. It looks strange when you have to reference the first element.

Reply:
That was exactly I had before. From SDK side, Freddy said they prefer the unified way, like what we did in ONS queries.